sources/prairie-dog/src/main.rs

```diff
@@ -0,0 +1,431 @@
+    let boot_partition_path = &state.active_set().boot;
+    let flags = nix::mount::MsFlags::MS_RELATIME
+        | nix::mount::MsFlags::MS_NOSUID
+        | nix::mount::MsFlags::MS_NOATIME;
+
+    info!(
+        "Mounting {} in {}",
+        boot_partition_path.display(),
+        LOCAL_BOOT_MOUNT_PATH
+    );
+    // Mount the active boot partition in /local/boot
+    nix::mount::mount(
+        Some(boot_partition_path),
+        LOCAL_BOOT_MOUNT_PATH,
+        Some("ext4"),
+        nix::mount::MsFlags::MS_RDONLY | flags,
+        Some(BOOT_MOUNT_SELINUX_CONTEXT),
+    )
+    .context(error::Mount {
+        path: LOCAL_BOOT_MOUNT_PATH,
+    })?;
+
+    info!("Creating bind mount in {}", BOOT_MOUNT_PATH);
+    // Create a bind mount from /local/boot to /boot for consistency with
+    // /var and /opt
+    nix::mount::mount(
+        Some(LOCAL_BOOT_MOUNT_PATH),
+        BOOT_MOUNT_PATH,
+        NONE,
+        nix::mount::MsFlags::MS_BIND | flags,
+        NONE,
+    )
+    .context(error::Mount {
+        path: BOOT_MOUNT_PATH,
+    })?;
+
+    // Make the mount points private so new mount namespaces don't have
+    // access to it. This has to be set as a different call otherwise
+    // the mount syscall returns `EINVAL`
+    make_private(LOCAL_BOOT_MOUNT_PATH)?;
+    make_private(BOOT_MOUNT_PATH)?;
+
+    Ok(())
+}
+
+// Sets the mount point's propagation as private
+fn make_private<S>(path: S) -> Result<()>
+where
+    S: AsRef<str>,
+{
+    let path = path.as_ref();
+    nix::mount::mount(NONE, path, NONE, nix::mount::MsFlags::MS_PRIVATE, NONE)
+        .context(error::SetupMount { path: path })
+}
+
+/// Loads the crash kernel using kexec-tools
+fn load_crash_kernel() -> Result<()> {
+    let kexec_crash_size_path = Path::new(KEXEC_CRASH_SIZE);
+    let kexec_crash_size = fs::read(kexec_crash_size_path).context(error::ReadFile {
+        path: kexec_crash_size_path,
+    })?;
+    let memory_allocated = String::from_utf8_lossy(&kexec_crash_size);
+
+    // We provide a more useful message when no memory was reserved for the crash kernel.
+    // This could happen when no sufficient memory is available in the host.
+    if memory_allocated.trim() == "0" {
+        return error::CrashKernelReservedMemory.fail();
+    }
+
+    let kexec_load_disabled_path = Path::new(KEXEC_LOAD_DISABLED);
+    let kexec_load_disabled_value =
+        fs::read(kexec_load_disabled_path).context(error::ReadFile {
+            path: kexec_load_disabled_path,
+        })?;
+    let kexec_load_disabled = String::from_utf8_lossy(&kexec_load_disabled_value).trim() == "1";
+
+    // We provide a more useful message when `kexec_load_disabled` is set to 1
+    if kexec_load_disabled {
+        return error::KexecLoadDisabled.fail();
+    }
+
+    info!("Loading crash kernel");
+    // Load the panic kernel from `BOOT_MOUNT_PATH`, letting kexec decide which syscall
+    // it should it use (KEXEC_LOAD and KEXEC_FILE_LOAD).
```

Comment:
Can we force it to always use `KEXEC_FILE_LOAD`?

Reply:
Yes we can, but for some reason the 5.4 kernel fails to use the `kexec_file_load` syscall, and it returns an `ENOSUP` error. The problem is not in the `kexec` library but rather in the kernel, so I need to debug the syscall to check what's returning the error. I'm already working with the kernel folks on this. 

In the 5.10 kernel, kexec always uses `kexec_file_load`, so I think it is OK to use `-a` for now and let `kexec` decide what syscall to use, since we are only enabling kdump in variants that use the 5.10 kernel.

Reply:
Wouldn't we expect to enable this in other variants relatively soon?  I imagine people building their own images/variants might want to enable it as well.  Wouldn't want a nasty surprise.

Reply:
Even if we enable this for other variants, and people build their own images/variants, they will still be locked with the `kexec_load` syscall for the aarch64 5.4 kernel variants since the syscall for this aarch64 was implemented after the 5.4 version.